### PR TITLE
[IJ plugin] Inspection to suggest adding an introspection block

### PR DESCRIPTION
**Pull request by BoD**
https://github.com/apollographql/apollo-kotlin/assets/372852/e9bb1324-684b-48c1-a7fc-274fe2f6a0ad

### CODE REVIEW

File: intellij-plugin/src/main/resources/messages/ApolloBundle.properties [hunk 1]
```diff
@@ -121,5 +121,9 @@ inspection.unusedField.quickFix=Delete field
 inspection.apollo4Available.displayName=Apollo Kotlin 4 is available
 inspection.apollo4Available.reportText=Apollo Kotlin 4 is available
 inspection.apollo4Available.quickFix=Migrate to Apollo Kotlin 4
+inspection.missingIntrospection.displayName=No introspection or registry block
+inspection.missingIntrospection.reportText=No introspection or registry block
+inspection.missingIntrospection.quickFix.introspection=Add introspection block
+inspection.missingIntrospection.quickFix.registry=Add registry block
```
Comment (martinbonnin) [inline intellij-plugin/src/main/resources/messages/ApolloBundle.properties:127]: I'd vote for skipping over registry for now and focus on getting the endpoint configured. As said above, I'd say this is the biggest win. It just happens to be in an `introspection {}` block but we might want to evolve the API to just `endoint` or so.

```suggestion
inspection.missingIntrospection.displayName=GraphQL endpoint not configured
inspection.missingIntrospection.reportText=GraphQL endpoint not configured
inspection.missingIntrospection.quickFix.introspection=Add introspection block
```

File: intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/inspection/MissingIntrospectionInspection.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          if (serviceBlockExpression.statements.none { it is KtCallExpression && (it.getMethodName() == "introspection" || it.getMethodName() == "registry") }) {
+            holder.registerProblem(
+                expression.calleeExpression!!,
+                ApolloBundle.message("inspection.missingIntrospection.reportText"),
+                AddBlockQuickFix(
+                    name = "inspection.missingIntrospection.quickFix.introspection",
+                    blockToAdd = """
+                      introspection {
+                          endpointUrl.set("https://example.com/graphql")
+                          schemaFile.set(file("src/main/graphql/schema.graphqls"))
```
Comment (martinbonnin) [inline intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/inspection/MissingIntrospectionInspection.kt:36]: Add headers as well? It's easy to remove if not needed and makes the API discoverable?

```suggestion
                          endpointUrl.set("https://example.com/graphql")
                          headers.put("ApiKey", "1234567890abcdef")
                          schemaFile.set(file("src/main/graphql/schema.graphqls"))
```

File: intellij-plugin/src/main/resources/messages/ApolloBundle.properties (hunk 1 above)
Comment (BoD) [reply intellij-plugin/src/main/resources/messages/ApolloBundle.properties:127]: Makes sense, agreed!

**Review by martinbonnin — APPROVED** [review]
👍

**Pull request merged**